toolium/utils/dataset.py

```diff
@@ -611,7 +646,7 @@ def get_value_from_context(param, context):
         if isinstance(value, dict) and part in value:
             value = value[part]
         # evaluate if in an array, access is requested by index
-        elif isinstance(value, list) and part.isdigit() and int(part) < len(value):
+        elif isinstance(value, list) and part.lstrip('-+').isdigit() and int(part) < len(value):
```

Comment:
Si el índice negativo es igual a la longitud de la lista también vale:
```suggestion
        elif isinstance(value, list) and part.lstrip('-+').isdigit() and int(part) <= len(value):
```

Reply:
![image](https://github.com/user-attachments/assets/96203a3f-b2ef-4a01-b216-9d0adf9de9b6)

Reply:
teniendo en cuenta que las listas empiezan en 0, si se tratase de un igual seria un out of index

Reply:
Ah, ya, estaba pensando en los valores negativos, como comentaba arriba. Va a haber que distinguir entre valores positivos y negativos entonces, porque sí que quiero poder hacer value[-1] en un array de longitud 1.

Se puede hacer así, por ejemplo:
```suggestion
        elif isinstance(value, list) and part.lstrip('-+').isdigit() \
                and int(part) < (len(value) + 1 if part.startswith("-") else len(value)):
```

También añadiría un assert accediendo a la posición -3 del array de la prueba `test_a_context_param_list_correct_negative_index()` que ya creaste, para probar el caso frontera.

Reply:
tal cual esta, ya puedes hacer eso, sin necesidad de poner el =

Reply:
Perdón, la segunda línea debería empezar por `and abs(int(part)) < ...`, que si no no se controla el rango de los índices negativos.